### ISSUE TITLE
What is the effect of using random_seed in fine-tuning that is different from pre-training

### ISSUE THREAD

**Issue by iivalchev** [LennartPurucker/finetune_tabpfn_v2#6]
As of now in the fine-tuning scripts a different random_seed is used (and can be changed when starting fine-tuning). However I am puzzled should we use different than the pre-training value. My questions rises by looking in to the positional encoding code in which the random_seed plays significant role.

For example if we use `feature_positional_embedding == "subspace"` isn't changing the seed to some extend discarding the pre-trained `feature_positional_embedding_embeddings` layer?

**Comment by LennartPurucker** [comment]
The seed for fine-tuning also controls the behavior of the fine-tuning itself; thus, it can have major effects due to data splitting or optimizer noise.

As far as I know, the `feature_positional_embedding` only needs to be consistent for a forward pass (and KV caching) and might not even have a significant effect anyhow. But I definitely would need to verify that again. 

I think it mainly dictates the embedding to be consistent across KV caching (https://github.com/PriorLabs/TabPFN/blob/9f208b70768bda1fe08b136842f3989b35b25081/src/tabpfn/model/transformer.py#L730). @LeoGrin do you have a better explanation?

**Comment by LeoGrin** [comment]
Agree @LennartPurucker, the feature positional embedding only need to be the same for the train and test points, meaning we need the same seed during fit and predict when using the kv cache.
But this shouldn't be affected by the global random seed anyway, as all of this is done inside the `with isolate_torch_rng` context manager.

**Comment by iivalchev** [comment]
@LeoGrin so the goal of the positional encoding is just to differentiate between the features?

**Comment by LeoGrin** [comment]
@iivalchev yes as far as I understand!

**Comment by iivalchev** [comment]
@LeoGrin and @LennartPurucker thank you for clearing this up. Closing this.
